### ISSUE TITLE
inconsistent sampling for population by palette size

### ISSUE THREAD

**Issue by hwaynefair** [leeoniya/RgbQuant.js#27]
RgbQuant is awesome!

I am using it in an application for beginning artists (painters) - reducing an image to a minimal number of colors...etc.

The strange behavior I find is that when I get a palette size of 10 - orange (in this particular case) is one of the most predominant color - and green doesn't show in the top 10 colors.

But when I get a palette size of 15 - green (which is far more visually correct) is the predominant color (by population).

Can you give me any advice on this?

Thanks so much!
```js
var q = new RgbQuant({ colors: getSlice });
q.sample(ctx);
var color = q.palette(true, true);
```

![slice15](https://user-images.githubusercontent.com/5210784/41726391-a698340a-7537-11e8-850d-9ca29f3cd8ec.jpg)
![slice10](https://user-images.githubusercontent.com/5210784/41726392-a6ac5b56-7537-11e8-8416-fa1d1955e9a3.jpg)

**Comment by leeoniya** [comment]
hey @hwaynefair ,

Can you link the original image here?

RgbQuant doesnt do great on very low color palette sizes (< 16). You can read a bit more in https://github.com/leeoniya/RgbQuant.js/issues/12. I'm currently on vacation, but will take a look in about a week. In the mean time, try bumping up the `minHueCols` setting.

**Comment by hwaynefair** [comment]
Really appreciate your timely response (well - any response at all for that matter)!
I will do as you advised - and please find the original image attached.

FYI - my "objective" here is to "paint" the HTML canvas with around 10 colors that are not too close to one another in value - to serve as an "under-painting" on the "real" canvas on which the artist is painting....

Thanks again!
-wayne

![light44ea973d-68e6-455f-bf65-d9f243ffbda9](https://user-images.githubusercontent.com/5210784/41777301-73ed1d40-75f0-11e8-998e-89796aa89742.jpg)

**Comment by hwaynefair** [comment]
Leon - you are so kind to go the extra mile with me on this. I apologize for my very late response.... I failed to notice the notification on github - and have been very distracted from my art application for several weeks - 
I did quickly run the demo using my image you added there - and it works perfectly! 
I will dig further into my code and see if I can isolate the issue on my side...
Thanks again for your kindness!
-wayne
